bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/WarmupThingHandler.java

```diff
@@ -0,0 +1,112 @@
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+
+/**
+ * The {@link WarmupThingHandler} is a super class for Things related to the Bridge consolidating logic.
+ *
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class WarmupThingHandler extends BaseThingHandler {
+
+    protected @Nullable MyWarmupAccountHandler bridgeHandler;
+
+    public WarmupThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            bridgeHandler = (MyWarmupAccountHandler) bridge.getHandler();
+            if (bridgeHandler != null) {
+                bridgeHandler.refreshFromCache();
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType && bridgeHandler != null) {
+            bridgeHandler.refreshFromCache();
+        }
+    }
+
+    protected void refreshFromServer() {
+        if (bridgeHandler != null) {
+            bridgeHandler.refreshFromServer();
+        }
+    }
+
+    /**
+     *
+     * @param temperature value returned from the API as an Integer * 10. i.e. 215 = 21.5 degrees C
+     * @return the temperature as a {@link QuantityType}
+     */
+    protected State parseTemperature(@Nullable Integer temperature) {
+        return temperature != null ? new QuantityType<>(temperature / 10.0, SIUnits.CELSIUS) : UnDefType.UNDEF;
+    }
+
+    /**
+     *
+     * @param date value returned from the API in local time, formatted as yyyy-MM-dd HH:mm:ss
+     * @param timezone {@link ZoneId} valid timezone from API
+     * @return the DateTime as a {@link DateTimeType}
+     */
+    protected State parseDate(@Nullable String date, @Nullable String timezone) {
+        return date != null && timezone != null ? new DateTimeType(ZonedDateTime
+                .of(LocalDateTime.parse(date, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")), ZoneId.of(timezone)))
```

Comment:
You could make `DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")` a static final constant to save resources.

Reply:
Code removed as it was unused